### PR TITLE
fudge gas estimate to prevent incorrect lower estimates from geth making stuff fail

### PR DESCRIPTION
**Pull request by nuevoalex**

### CODE REVIEW

File: source/libraries/ContractInterfaces.ts [hunk 1]
```diff
@@ -903,6 +903,27 @@
     }
 
 
+    export class GethContract extends Contract {
```
Comment (MicahZoltu) [inline source/libraries/ContractInterfaces.ts]: Feels like this should not be included (just for testing)?

File: source/libraries/ContractDeployer.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ export class ContractDeployer {
         await this.uploadAllContracts();
         await this.initializeAllContracts();
         await this.whitelistTradingContracts();
-
+
```
Comment (MicahZoltu) [inline source/libraries/ContractDeployer.ts]: Surprised your editor let you add this indent, I believe the editorconfig settings are "no whitespace on blank lines"?

File: source/libraries/ContractInterfaces.ts (hunk 1 above)
Comment (nuevoalex) [reply source/libraries/ContractInterfaces.ts]: Ah this shouldn't be there.

File: source/libraries/ContractInterfaces.ts [hunk 3]
```diff
@@ -34,8 +34,9 @@
         protected async remoteCall(abi: AbiFunction, parameters: Array<any>, txName: String, sender?: string, gasPrice?: BN, attachedEth?: BN): Promise<void> {
             const from = sender || this.accountManager.defaultAddress;
             const data = encodeMethod(abi, parameters);
-            // TODO: remove `gas` property once https://github.com/ethereumjs/testrpc/issues/411 is fixed
-            const gas = await this.connector.ethjsQuery.estimateGas(Object.assign({ to: this.address, from: from, data: data }, attachedEth ? { value: attachedEth } : {} ));
+            let gas = await this.connector.ethjsQuery.estimateGas(Object.assign({ to: this.address, from: from, data: data }, attachedEth ? { value: attachedEth } : {} ));
+            // This is to address an observed bug while running against geth where occasionally gas estimates are lower than required
```
Comment (MicahZoltu) [inline source/libraries/ContractInterfaces.ts]: Once we have a repro case in a Geth GitHub issue, we should link to it in this comment so a future reader can know whether it is safe to delete this terrible code.

File: source/libraries/ContractDeployer.ts (hunk 2 above)
Comment (nuevoalex) [reply source/libraries/ContractDeployer.ts]: I'll add something

**Pull request merged**